efi/util/util.go

```diff
@@ -21,3 +25,28 @@ func ReadNullString(f io.Reader) []byte {
 	}
 	return ret
 }
+
+// Parse an efivar as a UTF-16 string.
+func ParseUtf16Var(data *bytes.Buffer) (string, error) {
+	// Read two bytes which we won't keep.
+	// These initial two bytes are attribtues and not part of the actual string.
+	_, err := data.Read(make([]byte, 2))
```

Comment:
A little bit of confusion still. Normally when we read an UEFI variable we'll pass it to `ReadEfivars` which is going to read the two leading bytes (https://github.com/Foxboron/go-uefi/blob/master/efi/attributes/attributes.go#L68-L71).

Because of this we can just remove the above code and the two leading bytes in the test case data :)

Reply:
I see. The API in this PR might not be the best then, it might be better to have a `ReadEfiVarAsString` function that does reading and parsing, and make the parsing function private. Thoughts?

Reply:
Generally there are probably more strings in the UEFI API which might not fit this particular format, two bytes and null-terminated string. Keeping it as-is and fixing the test case is more flexible and we can just keep writing high-level functions for the variables is better.  So NACK on that idea :smile: